### PR TITLE
Thread names support

### PR DESCRIPTION
**Pull request by MatkovIvan**
Add thread names to crash report.
Please note that only "pthread"s names will be reported. Cocoa's names won't be reported yet.

AB#68137

### CODE REVIEW

**Comment by Kieran2k15** [comment]
I am not an expert on this topic, but I did look at making the same change myself some time ago and was stopped by this warning:

tl;dr : this change may cause a deadlock during crash report generation


https://landonf.org/code/crashreporting

Async-Safety
Imagine that the application has just acquired a lock prior to crashing. If the crash reporter attempts to acquire the same lock, it will wait forever: the crashed thread is no longer running, and it will never release the lock. When a deadlock like this occurs on iOS, the application will appear unresponsive for 20+ seconds until the system watchdog terminates the process, or the user force quits the application, and no valid crash report will be written.

In my previous post, I touched on async-safety issues around Objective-C and re-entrantly running the user's code. Most crash reporters have moved away from those APIs, but have introduced new async-safety issues in the process.

**One of the common issues I found in all new reporters was reliance on the pthread(3) API to fetch thread information, including thread names.** These APIs are not async-safe, however, and will acquire a global lock in most cases -- including when fetching a thread's name via pthread_getname_np(3). The result is that if your code crashes while any thread is holding the pthread thread-list lock, the entire crash reporter will deadlock. Since the crash reporters suspend all threads during reporting, this can occur even if the pthread calls themselves do not crash, but rather, a thread just happened to be executing a pthread() call at the time a crash occured.

**Comment by MatkovIvan** [comment]
@Kieran2k15 thanks for explicitly pointed on it.

Yeah, I thought about it, and that's why it's just a draft. I guess it can be closed at this point so that it doesn't look like ongoing/planned changes.
